### ISSUE TITLE
Wrong stacktrace offset

### ISSUE THREAD

**Issue by moltar** [pinojs/pino-caller#23]
Related #14, in particular the following comment: https://github.com/pinojs/pino-caller/issues/14#issuecomment-588139462

```
❯ node -v
v12.16.0
```

The full stacktrace looks like this:

```
❯ node src/demo/logjs.js
Error:
    at Proxy.asJson (~/pkg/node_modules/pino-caller/index.js:16:17)
    at Proxy.write (~/pkg/node_modules/pino/lib/proto.js:160:28)
    at Proxy.LOG (~/pkg/node_modules/pino/lib/tools.js:57:21)
    at Object.<anonymous> (~/pkg/src/demo/logjs.js:4:12)
    at Module._compile (internal/modules/cjs/loader.js:1157:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1177:10)
    at Module.load (internal/modules/cjs/loader.js:1001:32)
    at Function.Module._load (internal/modules/cjs/loader.js:900:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47
{"level":30,"time":1586590281920,"pid":54171,"hostname":"foo.lan","caller":"Module._compile (internal/modules/cjs/loader.js:1157:30)","msg":"info1"}
```

I got that stacktrace by adding a `console.log` to `asJson` method:

```js
  function asJson(...args) {
    console.log(Error().stack) // <!-- ADDED THIS
    args[0] = args[0] || Object.create(null)
    args[0].caller = Error().stack.split('\n').filter(s => !s.includes('node_modules/pino') && !s.includes('node_modules\\pino'))[STACKTRACE_OFFSET].substr(LINE_OFFSET)
    return pinoInstance[asJsonSym].apply(this, args)
  }
```

Wondering if it'd be safe to simply filter out all of the lines that match `internal/modules/cjs/loader.js`, similarly how `node_modules/pino` lines are currently filtered out.

I can create a PR.

**Comment by moltar** [comment]
Could also use [clean-stack](https://www.npmjs.com/package/clean-stack), or a similar package to clean the stack before.

**Comment by phra** [comment]
can you share your repro code? thanks.

**Comment by moltar** [comment]
Here you go: https://github.com/moltar/pino-exp

```
❯ npm start

> pino-exp@1.0.0 start /Users/foo/experiments/pino-exp
> node index.js

{"level":30,"time":1586759462999,"pid":99106,"hostname":"roman.lan","caller":"Module._compile (internal/modules/cjs/loader.js:1157:30)","msg":"info1"}

~/experiments/pino-exp master
❯ node -v
v12.16.0

```

**Comment by ilyaya** [comment]
The result is correct if .error() is used but .info() is incorrect.

**Comment by ilyaya** [comment]
I created a PR here: https://github.com/pinojs/pino-caller/pull/24

Regarding my previous comment, I'm not sure anymore. error() wasn't working for me today.

**Comment by phra** [comment]
i have created a new PR here: https://github.com/pinojs/pino-caller/pull/26

i prefer using the direct offset in the stack trace than a regex.

can you check if this works for you so i can merge it and release the patch? @moltar @ilyaya 

```diff
@@ -2,7 +2,7 @@
 
 require('source-map-support/register')
 const NODEJS_VERSION = parseInt(process.version.slice(1).split('.')[0], 10)
-const STACKTRACE_OFFSET = NODEJS_VERSION && NODEJS_VERSION > 6 ? 2 : 3
+const STACKTRACE_OFFSET = NODEJS_VERSION && NODEJS_VERSION > 6 ? 0 : 1
 const LINE_OFFSET = 7
 const { symbols } = require('pino')
 const { asJsonSym } = symbols
@@ -14,7 +14,7 @@ function traceCaller (pinoInstance) {
 
   function asJson (...args) {
     args[0] = args[0] || Object.create(null)
-    args[0].caller = Error().stack.split('\n').filter(s => !s.includes('node_modules/pino') && !s.includes('node_modules\\pino'))[STACKTRACE_OFFSET].substr(LINE_OFFSET)
+    args[0].caller = Error().stack.split('\n').slice(2).filter(s => !s.includes('node_modules/pino') && !s.includes('node_modules\\pino'))[STACKTRACE_OFFSET].substr(LINE_OFFSET)
     return pinoInstance[asJsonSym].apply(this, args)
   }
```

**Comment by moltar** [comment]
@phra responded on the PR. Thank you!!

**Comment by phra** [comment]
published as `v3.0.2`.
